### ISSUE TITLE
Core Utils to their own repo?

### ISSUE THREAD

**Issue by rowanc1** [simpeg/simpeg#667]
This is a first pass list at some of the utilities that could be taken out of simpeg and put in a small repo (`matutils`? maybe that is too specific?). The thought is that this would enable us to pull out some of the components we are building into other smaller more targeted, easier to deploy repositories (`objfun`, `regularize`) and we could experiment with putting things like `dask` in from the beginning of that (which would support parallel).

Nice utils:
* sparse: sdiag, sdinv, speye, spzeros, kron3 (these are all small wrappers on scipy.sparse)
* array: mkvc, ndgrid, ind2sub, sub2ind (basically wrappers on numpy)
* convenience: Zero, Identity

To ultimately delete from simpeg:
* `setKwargs`: should be replaced by `properties`
* `timeIt`: should be replaced by something like `dask`

See also #453 and #218

**Comment by lheagy** [comment]
Agreed. These are all over in discretize at the moment, so we may want to move this issue over there. 

In addition, there are a few utils in the `coordutils` that might also be moved over.
